be/src/formats/parquet/column_reader.cpp

```diff
@@ -511,34 +585,18 @@ Status ColumnReader::create(const ColumnReaderOptions& opts, const ParquetField*
         RETURN_IF_ERROR(reader->init(field, std::move(key_reader), std::move(value_reader)));
         *output = std::move(reader);
     } else if (field->type.type == LogicalType::TYPE_STRUCT) {
-        // build tmp mapping for ParquetField
-        std::unordered_map<std::string, size_t> field_name_2_pos;
-        for (size_t i = 0; i < field->children.size(); i++) {
-            const auto& format_field_name = opts.case_sensitive
-                                                    ? field->children[i].name
-                                                    : boost::algorithm::to_lower_copy(field->children[i].name);
-            field_name_2_pos.emplace(format_field_name, i);
-        }
+        std::vector<int32> subfield_pos(col_type.children.size());
+        get_subfield_pos_with_pruned_type(*field, col_type, opts.case_sensitive, subfield_pos);
 
         std::vector<std::unique_ptr<ColumnReader>> children_readers;
         for (size_t i = 0; i < col_type.children.size(); i++) {
-            const std::string& formatted_subfield_name =
-                    opts.case_sensitive ? col_type.field_names[i]
-                                        : boost::algorithm::to_lower_copy(col_type.field_names[i]);
-
-            auto it = field_name_2_pos.find(formatted_subfield_name);
-            if (it == field_name_2_pos.end()) {
-                LOG(WARNING) << "Struct subfield name: " + formatted_subfield_name + " not found.";
-                // Put nullptr ColumnReader for non existed subfield, it will append_default when fill data from parquet.
+            if (subfield_pos[i] == -1) {
```

Comment:
add comment, like : `-1 means subfield not existed, we still need to emplace_back nullptr`